catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -73,8 +123,71 @@ public Metacard transform(InputStream input, String id)
         }
     }
 
+    private Metacard transformWithExtractors(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        try (FileBackedOutputStream fbos = new FileBackedOutputStream(1000000)) {
+            try {
+                IOUtils.copy(input, fbos);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            String plainText = null;
+            try (InputStream isCopy = fbos.asByteSource()
+                    .openStream()) {
+                Parser parser = new AutoDetectParser();
+                ContentHandler contentHandler = new ToTextContentHandler();
+                TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
+                        contentHandler);
+                tikaMetadataExtractor.parseMetadata(isCopy, new ParseContext());
+                plainText = contentHandler.toString();
```

Comment:
In other work dealing with parsing the content of documents, we have had concerns with the size of the document being too large to load into memory.  These concerns have often been brought up by @stustison .  What alleviates those concerns for when converting to plainText or creating the PDDocument?  

While I can understand GBs of plain text seems unlikely (million page pdf)... is the same true for PDDocument?  And even at unlikely, isn't that a vulnerability if we make the assumption.

I am sure there is an answer in here already I am just not seeing, and wanted to get clarification for other work being done, such as DDF-1881 @mweser is working on.